src/components/calculator/Calculator.jsx

```diff
@@ -1,80 +1,189 @@
 import React from 'react';
+import calculate from './logic/calculate';
 import './calculator.scss';
 
 export default class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      calcObj: { total: null, next: null, operation: null },
+      display: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleDisplayState(beforeCalcObj, afterCalcObj) {
+    // Handle display UI state
+    if (!afterCalcObj.operation && !afterCalcObj.next && afterCalcObj.total) {
+      this.setState({ display: afterCalcObj.total });
+    } else if (beforeCalcObj.operation && afterCalcObj.next) {
+      this.setState({ display: afterCalcObj.next });
+    } else if (afterCalcObj.total && afterCalcObj.operation) {
+      this.setState({ display: afterCalcObj.total });
+    } else if (!afterCalcObj.operation) {
+      this.setState({ display: afterCalcObj.next });
+    } else {
+      this.setState({ display: 0 });
+    }
+  }
+
+  handleClick(e) {
+    // Handle calculator logic state
+    const { calcObj } = this.state;
+    const newCalcObj = calculate(calcObj, e.target.innerText);
+    this.setState({ calcObj: newCalcObj });
+    this.handleDisplayState(calcObj, newCalcObj);
+  }
+
   render() {
+    const { display, calcObj } = this.state;
     return (
       <div className="calculator">
-        <div className="calculator__display">0</div>
+        <div className="calculator__display">
+          {display ?? calcObj.next ?? 0}
+        </div>
         <div className="calculator__keypad">
           <div className="calculator__keypad-row">
-            <button type="button" className="calculator__keypad-button">
+            <button
+              onClick={this.handleClick}
+              type="button"
+              className="calculator__keypad-button"
+            >
               AC
             </button>
-            <button type="button" className="calculator__keypad-button">
+            <button
+              onClick={this.handleClick}
+              type="button"
+              className="calculator__keypad-button"
+            >
```

Comment:
- [ ] Kindly create another component as a child of the `Calculator` component and pass it as data ass props to the child component. You could put all the buttons in an array, map through the array and pass each button as a prop to the component. You can read this article to have a better understanding https://www.geeksforgeeks.org/how-to-map-data-into-components-using-reactjs/

Reply:
Understood. Will do 👍

Reply:
Hi @Gambit142. Thanks for your feedback 💪. I was able to make a child component and render the buttons by sending its props and lifting the state to its parent 🤓

Reply:
Hi @hsztan, Great work, you did it better than I suggested. 👍